### ISSUE TITLE
Shift-clicking hotbar items moves to inventory, not chest

### ISSUE THREAD

**Issue by Heptagon-ru** [anegostudios/VintageStory-Issues#28]
1.5.2.2, the official server.
I have floatyGuis on.
When a chest is open and player inventory is closed, shift-clicking items from hotbar moves them into player's inventory, not to the open chest inventory.
Also iirc if chest has an uncomplete stack of the same item, the stack is filled, but the remains go to player inventory.
I would expect that if a single chest inventory is open, all hotbar items are moved there, and never to the closed player inventory.

**Comment by copygirl** [comment]
Yes, I find the shift-click moving to be a little annoying and, honestly, would rather use the shift key for other useful inventory activities, such as selecting multiple items at once, than to stumble over guessing where the player would want their items to go at any specific time.

I suppose just preferring an open chest over the inventory is probably a good way to go. Additionally, if one has multiple chests open, it should prefer chests that already have the same item in them.

**Comment by tyronx** [comment]
There was also a recent change in 1.5.2.2 that makes them always get moved into the backpack inventory even when closed (in order to fix another bug). But I can probably make it so that closed backpack inventory gets excluded again from shift+click moves. 

@copygirls suggestions also sound good.

**Comment by Heptagon-ru** [comment]
Actually there was a forum discussion which resulted in vOS creation. And the last-ish comment there is from @copygirl with some feedback on vOS with nice ideas:
https://www.vintagestory.at/forums/topic/275-improving-replacing-the-gui/?tab=comments#comment-1616
I love floatyGuis but also have some minor suggestions to make vOS experience ideal. Gonna add my suggestions there asap.
So maybe see if all inventory mechanics could be improved further, also including shift-clicking refactoring.

**Comment by copygirl** [comment]
I'm not sure. To me personally this is still an on-going issue. Shift clicking often doesn't do what I want it to do. Shift clicking your inventory moves items to your hotbar before moving them to a chest, still. But I'll leave it up to @tyronx. Maybe I'll create a issue later with my arguments layed out and offering an alternative?

**Comment by Heptagon-ru** [comment]
Yeah, the shift-clicking inventory moving items into hotbar is kinda annoying. Less annoying than the original issue somehow. I don't know if it is best to reopen this one or create a new one :) Let's burden @tyronx with this decision :D
